### ISSUE TITLE
FR: Add CLI option to delete a profile

### ISSUE THREAD

**Issue by sailorfrag** [tailscale/tailscale#12255]
### What are you trying to do?

When an account has been deleted from the coordination server, it can still be present on the client. There should be an easy way to remove it from the client.

### How should we solve this?

The localapi supports deleting a profile, but the CLI (and none of the GUIs as far as I can tell) does not have this connected to anything. It should be added, perhaps alongside `tailscale switch -list`, or perhaps somewhere else.

### What is the impact of not solving this?

Users are unable to clean up old profiles. This comes up from time to time in support requests and elsewhere. For example, on Reddit: https://www.reddit.com/r/Tailscale/comments/1bwgp9v/remove_accounts_listed_in_win_app/

**Comment by agottardo** [comment]
On the GUI side, iOS is the only client to support this by holding down on a profile to reveal a "Delete" option. I'll be adding this to macOS.

**Comment by jimmybrancaccio** [comment]
@agottardo I think this already exists in the macOS client. If the account is logged out a Remove Account... button shows up.

<img width="712" alt="Screenshot 2024-05-27 at 16 08 57" src="https://github.com/tailscale/tailscale/assets/405585/f7882407-79e5-4041-9e2f-a7f6b6b32c36">

**Comment by dorinclisu** [comment]
In the meantime, any way to manually delete profiles on linux?

**Comment by Esteban-Bermudez** [comment]
> In the meantime, any way to manually delete profiles on linux?

I have just submitted a PR that should fix this issue

**Comment by Esteban-Bermudez** [comment]
This appears to be closed, but the solution has not yet been merged. My first PR for this fixed a bug with the delete profile URL endpoint, but then I submitted PR #15150 to add the cli subcommand that is needed for this to be resolved.

Currently, Linux users have no way to remove accounts

**Comment by yoyoma2** [comment]
> This appears to be closed, but the solution has not yet been merged.

Agreed this issue still exists at least on Windows and shouldn't be closed. This [ugly workaround](https://www.reddit.com/r/Tailscale/comments/1bwgp9v/comment/l7i8akz/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button) on Windows does indeed delete all profiles.

**Comment by bradfitz** [comment]
@yoyoma2, https://github.com/tailscale/tailscale/commit/1c9aaa444da163bf0597cef09a100a4e7a0221b8 was merged. It'll be in the next (non-patch) release.
